pkg/scaffold/helm/chart.go

```diff
@@ -265,3 +270,116 @@ func createChartFromRemote(destDir string, opts CreateChartOptions) (*chart.Char
 
 	return createChartFromDisk(destDir, chartArchive, false)
 }
+
+func GenerateRoleRules(chart *chart.Chart) ([]rbacv1.PolicyRule, []error) {
+	helmOperatorRules := []rbacv1.PolicyRule{
+		// We need this rule so tiller can read namespaces to ensure they exist
+		{
+			APIGroups: []string{""},
+			Resources: []string{"namespaces"},
+			Verbs:     []string{"get"},
+		},
+
+		// We need this role for leader election to work
+		{
+			APIGroups: []string{""},
+			Resources: []string{"configmaps"},
+			Verbs:     []string{"*"},
+		},
+	}
+	customRules, errs := parseRulesFromChart(chart)
+	customRules = append(customRules, helmOperatorRules...)
+	return customRules, errs
+}
+
+func parseRulesFromChart(chart *chart.Chart) ([]rbacv1.PolicyRule, []error) {
+	var errs []error
+	groups := map[string]map[string]struct{}{}
+	gotmplRegex := regexp.MustCompile(`{{.*}}`)
+
+	for _, tmpl := range chart.GetTemplates() {
+		data := string(tmpl.GetData())
+		name := tmpl.GetName()
+		yamlTmpls := strings.Split(data, "\n---\n")
+
+		for i, yamlTmpl := range yamlTmpls {
+			var apiVersion, kind string
+
+			clean := func(s string) string {
+				s = strings.TrimSpace(s)
+				s = strings.Trim(s, "'\"")
+				return s
+			}
+
+			// Extract the apiVersion and kind from the template
+			lines := strings.Split(yamlTmpl, "\n")
+			for _, line := range lines {
+				line = strings.TrimSpace(line)
+				if strings.HasPrefix(line, "apiVersion") {
+					apiVersion = clean(strings.TrimPrefix(line, "apiVersion:"))
+				} else if strings.HasPrefix(line, "kind:") {
+					kind = clean(strings.TrimPrefix(line, "kind:"))
+				}
+			}
+
+			// Skip templates that don't look like Kubernetes resources
+			if apiVersion == "" && kind == "" {
```

Comment:
What if we just do:
```go
if apiVersion == "" || kind == "" {
        errs = append(errs, fmt.Errorf("could not find resource kind or apiVersion %s[%d]", name, i))
	continue
}
```

Reply:
This has also been slightly reworked. I left the separate checks for `apiVersion` and `kind` so that the logged warning would be more specific.